spec.emu

```diff
@@ -895,27 +959,21 @@
           1. If _state_.[[GeneratedColumn]] &lt; 0, then
             1. Optionally report an error.
             1. Return.
+          1. Let _generatedPosition_ be a new Position Record { [[Line]]: _state_.[[GeneratedLine]], [[Column]]: _state_.[[GeneratedColumn]] }.
           1. Perform DecodeMappingsField of |OriginalSource| with arguments _state_, _mappings_, _names_ and _sources_.
-          1. If _state_.[[SourceIndex]] &lt; 0 or _state_.[[SourceIndex]] ≥ the number of elements of _sources_, then
-            1. Optionally report an error.
-            1. Let _source_ be *null*.
-          1. Else, let _source_ be _sources_[_state_.[[SourceIndex]]].
           1. Perform DecodeMappingsField of |OriginalLine| with arguments _state_, _mappings_, _names_ and _sources_.
-          1. If _state_.[[OriginalLine]] &lt; 0, then
-            1. Optionally report an error.
-            1. Let _originalLine_ be *null*.
-          1. Else, let _originalLine_ be _state_.[[OriginalLine]].
           1. Perform DecodeMappingsField of |OriginalColumn| with arguments _state_, _mappings_, _names_ and _sources_.
-          1. If _state_.[[OriginalColumn]] &lt; 0, then
+          1. If _state_.[[SourceIndex]] &lt; 0 or _state_.[[SourceIndex]] ≥ the number of elements of _sources_ or _state_.[[OriginalLine]] &lt; 0 or _state_.[[OriginalColumn]] &lt; 0, then
             1. Optionally report an error.
-            1. Let _originalColumn_ be *null*.
-          1. Else, let _originalColumn_ be _state_.[[OriginalColumn]].
+            1. Let _originalPosition_ be *null*.
+          1. Else,
+            1. Let _originalPosition_ be a new Original Position Record { [[SourceIndex]]: _sources_[_state_.[[SourceIndex]]], [[Line]]: _state_.[[OriginalLine]], [[Column]]: _state_.[[OriginalColumn]] }.
```

Comment:
Should this be `[[SourceIndex]]: _state_.[[SourceIndex]]` here instead? (since it's the index and not the record now)

Reply:
Great catch! I fixed it for now as you suggested, but do you think that the "Original Position Record" should just keep an index as it is now, or should it resolve to the actual "Decoded Source Record"?

Reply:
Hmm, that's a good question, I'm not sure. I think it could be made to work with an index too but actually it does require another change, which is that for index maps we would need to renumber the indices (since they are relative to the sub-source map).

@nicolo-ribaudo do you have any opinion on this?

Reply:
I don't have a strong preference in either direction.